Sources/MlemMiddleware/API Client/ApiClient.swift

```diff
@@ -159,6 +160,30 @@ public class ApiClient {
     }
     
     private func execute(_ urlRequest: URLRequest) async throws -> (Data, URLResponse) {
+        var urlRequest: URLRequest = urlRequest // make mutable
+        
+        // add 0.18x "auth" param if on 18.x (or unrecognized) instance
+        let apiVersionRecognized: Bool
+        if case .other = fetchedVersion {
+            apiVersionRecognized = false
+        } else {
+            apiVersionRecognized = true
+        }
```

Comment:
*nit*
Could this logic be stored as a computed property of `SiteVersion` maybe? 

```swift
var requiresAuthFieldInRequestBody: Bool {
   self == .other || self < .v19_0
}
```

Then the `if` statement on line 174 would be:

```swift
if let token, fetchedVersion.requiresAuthFieldInRequestBody ?? true, urlRequest.httpMethod != "GET" {
```

(Tbh we should probably have quite a few computed properties on `SiteVersion` for various features - `allowsBatchImageUploads` could go there too rather than on `ApiClient`. OOS though)